include/ttmlir/Target/Utils/MLIRToFlatbuffer.h

```diff
@@ -136,6 +137,35 @@ inline ::tt::target::DataType toFlatbuffer(FlatbufferObjectCache &,
   }
 }
 
+inline ::flatbuffers::Optional<::tt::target::DataType>
+toFlatbufferOptional(FlatbufferObjectCache &cache,
+                     ::std::optional<::mlir::tt::DataType> dataType) {
+  return dataType.has_value() ? ::flatbuffers::Optional<::tt::target::DataType>(
+                                    toFlatbuffer(cache, dataType.value()))
+                              : ::flatbuffers::nullopt;
+}
+
+inline ::tt::target::TensorLayout toFlatbuffer(FlatbufferObjectCache &cache,
+                                               ttnn::Layout layout) {
+  switch (layout) {
+  case ttnn::Layout::RowMajor:
+    return ::tt::target::TensorLayout::RowMajor;
+  case ttnn::Layout::Tile:
+    return ::tt::target::TensorLayout::Tile;
+  case ttnn::Layout::Invalid:
```

Comment:
Do you know if this enum value is used ever?

Reply:
Don't really know, but for the sake of completeness and future-proofing, decided to include it.